File: gman-experiments/image_cropper/crop.py
import cv2

def main():
    # Read the input image
    image = cv2.imread("test_image.jpg")

    # Check if the image was successfully loaded
    if image is None:
        print("Error: Couldn't open or find the image.")
        return

    # Define the cropping region
    x = (image.shape[1] - 724) // 2  # Center crop for width
    y = (image.shape[0] - 715) // 2  # Center crop for height
    width = 724
    height = 715

    # Crop the image
    cropped_image = image[y:y+height, x:x+width]

    # Save the cropped image
    cv2.imwrite("cropped_image.jpg", cropped_image)

    print("Cropped image saved successfully.")

if __name__ == "__main__":
    main()



#replace
# height 2749px , width 2769px

File: gman-experiments/image_cropper/crop.cpp
#include <opencv2/opencv.hpp>
#include <iostream>

using namespace cv;
using namespace std;

int main() {
    // Read the input image
    Mat image = imread("input_image.jpg");
    
    // Check if the image was successfully loaded
    if(image.empty()) {
        cout << "Error: Couldn't open or find the image." << endl;
        return -1;
    }

    // Define the cropping region
    int x = (image.cols - 724) / 2; // Center crop for width
    int y = (image.rows - 715) / 2; // Center crop for height
    int width = 724;
    int height = 715;

    // Crop the image
    Rect cropRegion(x, y, width, height);
    Mat croppedImage = image(cropRegion);

    // Save the cropped image
    imwrite("cropped_image.jpg", croppedImage);

    cout << "Cropped image saved successfully." << endl;

    return 0;
}
